article-analyzer/src/main/java/com/reader/article_analyzer/ArticleAnalyzerApplication.java

```diff
@@ -0,0 +1,29 @@
+package com.reader.article_analyzer;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.ConfigurableApplicationContext;
+
+import java.io.IOException;
+
+
+@SpringBootApplication
+public class ArticleAnalyzerApplication {
+
+    public static void main(String[] args) throws IOException {
+        ConfigurableApplicationContext context = SpringApplication.run(ArticleAnalyzerApplication.class, args);
+
+        ArticleCategoryAnalyzer analyzer = context.getBean(ArticleCategoryAnalyzer.class);
+        ArticleCategorySaver saver = context.getBean(ArticleCategorySaver.class);
+        UserNotifier userNotifier = context.getBean(UserNotifier.class);
+
+        analyzer.analyzeArticleContent("articles.json", 0);
+        analyzer.analyzeArticleContent("articles.json", 1);
+        analyzer.analyzeArticleContent("articles.json", 2);
+
+        saver.JsonSaver();
+
+        userNotifier.notifyUsersAboutCategoryChanges();
```

Comment:
move all this business logic into some class, here only should be
```suggestion
analyzer.analyze()
```

Reply:
Analyzer analyzer = context.getBean(Analyzer.class);
        analyzer.analyze();

Reply:
In class ArticleAnalyzerApplication

Reply:
```suggestion

import org.springframework.stereotype.Component;

import java.io.IOException;

@Component
public class Analyzer {
    private final ArticleCategoryAnalyzer articleCategoryAnalyzer;
    private final ArticleCategorySaver articleCategorySaver;
    private final UserNotifier userNotifier;


    public Analyzer(ArticleCategoryAnalyzer articleCategoryAnalyzer, ArticleCategorySaver articleCategorySaver, UserNotifier userNotifier) {
        this.articleCategoryAnalyzer = articleCategoryAnalyzer;
        this.articleCategorySaver = articleCategorySaver;
        this.userNotifier = userNotifier;
    }

    public void analyze() throws IOException {
        articleCategoryAnalyzer.analyzeArticleContent("articles.json", 0);
        articleCategoryAnalyzer.analyzeArticleContent("articles.json", 1);
        articleCategoryAnalyzer.analyzeArticleContent("articles.json", 2);
        articleCategorySaver.JsonSaver();
        userNotifier.notifyUsersAboutCategoryChanges();
    }
}

```

Reply:
import org.springframework.stereotype.Component;

import java.io.IOException;

@Component
public class Analyzer {
    private final ArticleCategoryAnalyzer articleCategoryAnalyzer;
    private final ArticleCategorySaver articleCategorySaver;
    private final UserNotifier userNotifier;


    public Analyzer(ArticleCategoryAnalyzer articleCategoryAnalyzer, ArticleCategorySaver articleCategorySaver, UserNotifier userNotifier) {
        this.articleCategoryAnalyzer = articleCategoryAnalyzer;
        this.articleCategorySaver = articleCategorySaver;
        this.userNotifier = userNotifier;
    }

    public void analyze() throws IOException {
        articleCategoryAnalyzer.analyzeArticleContent("articles.json", 0);
        articleCategoryAnalyzer.analyzeArticleContent("articles.json", 1);
        articleCategoryAnalyzer.analyzeArticleContent("articles.json", 2);
        articleCategorySaver.JsonSaver();
        userNotifier.notifyUsersAboutCategoryChanges();
    }
}

Reply:
I changed it and created a new class Analyzer with business logic